src/IoTHub.Portal.Infrastructure/Services/LoRaWanManagementService.cs

```diff
@@ -10,15 +10,20 @@ namespace IoTHub.Portal.Infrastructure.Services
     using System.Text.Json;
     using System.Threading.Tasks;
     using IoTHub.Portal.Application.Services;
+    using IoTHub.Portal.Domain.Options;
     using IoTHub.Portal.Models.v10.LoRaWAN;
+    using Microsoft.Extensions.Options;
 
     internal class LoRaWanManagementService : ILoRaWanManagementService
     {
         private readonly HttpClient httpClient;
 
-        public LoRaWanManagementService(HttpClient httpClient)
+        public LoRaWanManagementService(HttpClient httpClient, IOptions<LoRaWANOptions> loRaWANOptions)
         {
             this.httpClient = httpClient;
+            this.httpClient.BaseAddress = new Uri(loRaWANOptions?.Value.KeyManagementUrl);
+            this.httpClient.DefaultRequestHeaders.Add("x-functions-key", loRaWANOptions?.Value.KeyManagementCode);
+            this.httpClient.DefaultRequestHeaders.Add("api-version", "2022-03-04");
```

Comment:
This should be modified directly on the startup (the client is injected with the correct settings).

Maybe adding this into appsettings...

Reply:
@lucas-mrq can you also add the api version as an option? 
Please update your proposal to make those changes in the program startup instead of during the construction.

Reply:
I added is.

Something similar is added into the Startup file but is never loaded. To fix it we use the same process into LoRaWanManagementService.cs

Reply:
This is done at: https://github.com/CGI-FR/IoT-Hub-Portal/blob/9d0170feddadda776163882f353f4d3ae170adba/src/IoTHub.Portal.Infrastructure/Startup/AzureServiceCollectionExtension.cs#L73C13-L80C15

Please verify your assumption but I'm pretty sure that it should work without your changes.

Alternatively, give me a screenshot of what your are getting in the httpClient default headers (take care of the api-key) please.

Reply:
This is what we go during LoraWanManagement initialization:

![image](https://github.com/CGI-FR/IoT-Hub-Portal/assets/39192032/4fce89b8-706a-467f-8f33-c9cd87a7c741)
![image](https://github.com/CGI-FR/IoT-Hub-Portal/assets/39192032/2c43db1b-93a4-464e-8cc3-188915631aec)

Reply:
Ok, so the bug should be fixed in the startup, not this service.

I'll try to find why